vajram/vajram-codegen/src/main/java/com/flipkart/krystal/vajram/codegen/VajramCodeGenerator.java

```diff
@@ -49,140 +65,696 @@
+            buildResolverForMultipleBindings(method, inputs, bindParamName, ifBlockBuilder);
+        } else {
+            buildResolverForSingleBindings(method, inputs, parameter, ifBlockBuilder);
+        }
+        ifBlockBuilder.endControlFlow();
+        return ifBlockBuilder;
     }
 
-    builderClass.addMethod(
-        // public Request build(){
-        //   return new Request(input1, input2, input3)
-        // }
-        methodBuilder("build")
-            .returns(ClassName.get(packageName, requestClassName))
-            .addModifiers(PUBLIC)
-            .addStatement(
-                "return new %s(%s)".formatted(requestClassName, String.join(", ", inputNames)))
-            .build());
-    StringWriter writer = new StringWriter();
-    FromAndTo fromAndTo =
-        fromAndToMethods(
-            inputDefs.stream()
-                .filter(
-                    inputDef -> inputDef.toInputDefinition().sources().contains(InputSource.CLIENT))
-                .toList(),
-            ClassName.get(packageName, requestClassName));
-    try {
-      JavaFile.builder(
-              packageName,
-              requestClass
-                  .addMethod(requestConstructor.build())
-                  .addMethod(fromAndTo.from())
-                  .addMethod(fromAndTo.to())
-                  .addType(builderClass.build())
-                  .build())
-          .build()
-          .writeTo(writer);
-    } catch (IOException ignored) {
+    /**
+     * Method to generate resolver code for variables having single resolver.
+     * @param method Resolve method
+     * @param inputs Resolve inputs
+     * @param parameter Resolve method parameter
+     * @param ifBlockBuilder {@link CodeBlock.Builder}
+     */
+    private void buildResolverForSingleBindings(Method method, String[] inputs, Parameter parameter,
+            CodeBlock.Builder ifBlockBuilder) {
+        String variableName = parameter.getName();
+        boolean controlFLowStarted = false;
+        // Identify resolve method return type
+        final Type returnType = method.getReturnType();
+        if (method.getReturnType() == DependencyCommand.class) {
+            // Method returns DependencyCommand so add code block for skip() operation
+            final Type actualTypeArgument = ((ParameterizedType) method.getGenericReturnType()).getActualTypeArguments()[0];
+            ifBlockBuilder.addStatement("DependencyCommand<$T> $L = super.$L(inputs.getInputValueOrThrow($S))",
+                    actualTypeArgument, variableName, method.getName(), parameter.getAnnotation(BindFrom.class).value());
+            ifBlockBuilder.beginControlFlow("if($L instanceof DependencyCommand.Skip<$T> skip)",
+                    parameter.getName(), actualTypeArgument);
+            ifBlockBuilder.addStatement("\t return skip.cast()");
+            ifBlockBuilder.add("} else {\n\t");
+            controlFLowStarted = true;
+        } else {
```

Comment:
Check return type iterable for fanout use case (DependencyCommand.multiExecuteWith)

Reply:
Done